base_layer/core/src/mempool/sync_protocol/mod.rs

```diff
@@ -174,6 +182,55 @@ where TSubstream: AsyncRead + AsyncWrite + Unpin + Send + Sync + 'static
         }
     }
 
+    async fn spawn_handle_block_event(&mut self, block_event: &BlockEvent) {
```

Comment:
total nit :P - I would say this function just handles a block event, the fact that the protocol needs to be spawned is an implementation detail. 
```suggestion
    async fn handle_block_event(&mut self, block_event: &BlockEvent) {
```